qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/IndexingIT.java

```diff
@@ -285,7 +285,7 @@ public void testUpdateSnapshotStatus() throws Exception {
 
     public void testSyncedFlushTransition() throws Exception {
         Nodes nodes = buildNodeAndVersions();
-        assumeTrue("bwc version is on 7.x", nodes.getBWCVersion().before(Version.V_8_0_0));
+        assumeTrue("bwc version is on 7.x", BWC_NODES_VERSION.startsWith("7."));
```

Comment:
This sort of check is not ideal - is this something to replace with a feature check?

Reply:
I agree it's not ideal; as I wrote in the description, I want to give it another pass after we have something to do feature-based testing (see also the various `isOriginalClusterVersionAtLeast`, the version test in `SqlSearchIT`, etc.). 
If we agree we like https://github.com/elastic/elasticsearch/pull/100811, I can introduce the same/similar concept here (since these are BwC tests, they suffer from the same problem that FeatureService will not be directly accessible - and not present at all for older versions) and come back later to this PR to replace all version-based checks with feature-based checks

Reply:
This however was particularly nasty, so I rewrote it to make clear already that it is a  feature check (version-based, for now)